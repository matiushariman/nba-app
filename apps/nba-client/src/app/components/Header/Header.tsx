import GithubLogo from './assets/github-mark-white.svg';

const Header = () => (
  <div className="dark:bg-[#000] text-white pl-8 pr-8 pt-3 pb-3 flex justify-between items-center">
    <img
      width="48"
      height="28"
      src="https://cdn.nba.com/logos/leagues/logo-nba.svg"
      alt="NBA Logo"
    />
    <a
      href="https://github.com/matiushariman/nba-app"
      target="_blank"
      className="text-sm"
      rel="noreferrer"
    >
      <img src={GithubLogo} alt="github logo" width={24} height={24} />
    </a>
  </div>
);

export default Header;
