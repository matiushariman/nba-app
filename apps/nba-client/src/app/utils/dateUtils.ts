import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
