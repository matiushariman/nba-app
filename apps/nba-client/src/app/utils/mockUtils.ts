export const shouldUseMock = process.env['NODE_ENV'] === 'development';
