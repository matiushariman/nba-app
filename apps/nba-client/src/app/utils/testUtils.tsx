import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
