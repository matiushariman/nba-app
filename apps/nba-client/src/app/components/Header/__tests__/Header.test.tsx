import Header from '..';

import { render } from '../../../utils/testUtils';

describe('components/Header', () => {
  it('should render Header that matches snapshot', () => {
    const { baseElement } = render(<Header />);

    expect(baseElement).toMatchSnapshot();
  });
});
