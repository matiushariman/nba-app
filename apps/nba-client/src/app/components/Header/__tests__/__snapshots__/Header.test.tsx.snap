// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header should render Header that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="dark:bg-[#000] text-white pl-8 pr-8 pt-3 pb-3"
    >
      <img
        alt="NBA Logo"
        height="28"
        src="https://cdn.nba.com/logos/leagues/logo-nba.svg"
        width="48"
      />
    </div>
  </div>
</body>
`;
